src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -22323,6 +22323,194 @@ public void ScopedReserved_Alias_Escaped()
                 );
         }
 
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/75828")]
+        public void UnscopedRefAttribute_RefSafetyRules_ThisParameter()
+        {
+            var source = """
+                using System.Diagnostics.CodeAnalysis;
+                struct S
+                {
+                    public int F;
+
+                    [UnscopedRef] public ref int Ref() => ref F;
+                }
+                """;
+            CreateCompilation([source, UnscopedRefAttributeDefinition],
+                parseOptions: TestOptions.Regular10).VerifyDiagnostics(
+                // (6,47): error CS8170: Struct members cannot return 'this' or other instance members by reference
```

Comment:
>error CS8170: Struct members cannot return 'this' or other instance members by reference [](http://example.com/codeflow?start=27&length=88)

This looks like a breaking change, although it also looks like users of this library would ignore the `[UnscopedRef]` attribute on the method since the assembly is not marked with `[RefSafetyRules]`, so perhaps the breaking change when compiling this library is appropriate.

Consider adding a note to the Breaking Change document.

Is this the only breaking change in the PR? (From the tests below, it looks like in other cases we're reporting other errors where `[UnscopedRef]` is used (and ignored) with C#10 rules.)

Reply:
Should the compiler report a diagnostic when `[UnscopedRef]` is applied to a source method that is compiled with earlier ref safety rules? Otherwise, the library author may not realize the annotation will be ignored by callers. For instance, would the following still compile with earlier ref safety rules without warnings or errors?
```csharp
struct S
{
    int F;
    [UnscopedRef] public ref int Ref() => ref Unsafe.AsRef(in F);
}
static class C
{
    static ref int M()
    {
        S s = default;
        return ref s.Ref();
    }
}
```

Reply:
cc @jaredpar, @RikkiGibson

Reply:
It feels like an oversight that we did not report an error when UnscopedRef was used in presence of the old ref safety rules.

Reply:
> For instance, would the following still compile with earlier ref safety rules without warnings or errors?

No, it wouldn't compile. There will be errors because you are still using `this` by reference inside `S.Ref` which is disallowed there.